evennia/utils/opclasses.py

```diff
@@ -0,0 +1,319 @@
+        valid: Shortcut to the loaded VALID_HANDLER.
+        valid_type (str): The key of the Validator this uses.
+    """
+    expect_type = ''
+    valid = _VAL
+    valid_type = ''
+
+    def __str__(self):
+        return self.key
+
+    def __init__(self, handler, key, description, default, save_data=None):
+        """
+
+        Args:
+            handler (OptionHandler): The OptionHandler that 'owns' this Option.
+            key (str): The name this will be used for storage in a dictionary. Must be unique per
+                OptionHandler.
+            description (str): What this Option's text will show in commands and menus.
+            default: A default value for this Option.
+            save_data: Whatever was saved to Attributes. This differs by Option.
+        """
+        self.handler = handler
+        self.key = key
+        self.default_value = default
+        self.description = description
+        self.save_data = save_data
+
+        # Value Storage contains None until the Option is loaded.
+        self.value_storage = None
+
+        # And it's not loaded until it's called upon to spit out its contents.
+        self.loaded = False
+
+    def load(self):
+        """
+        Takes the provided save data, validates it, and gets this Option ready to use.
+
+        Returns:
+            Boolean: Whether loading was successful.
+        """
+        if self.save_data is not None:
+            try:
+                self.value_storage = self.valid_save(self.save_data)
+                self.loaded = True
+                return True
+            except Exception as e:
+                print(e)  # need some kind of error message here!
+        return False
+
+    def customized(self):
+        return self.value_storage != self.default_value
+
+    def valid_save(self, save_data):
+        """
+        Perform sanity-checking on the save data. This isn't the same as Validators, as Validators deal with
+        user input. save data might be a timedelta or a list or some other object. isinstance() is probably
+        very useful here.
+
+        Args:
+            save_data: The data to check.
+
+        Returns:
+            Arbitrary: Whatever the Option needs to track, like a string or a datetime. Not the same as what
+                users are SHOWN.
+        """
+        return save_data
+
+    def clear(self):
+        """
+        Resets this Option to default settings.
+
+        Returns:
+            self. Why?
+        """
+        self.value_storage = None
+        self.loaded = False
+        return self
+
+    @property
+    def default(self):
+        return self.default_value
+
+    @property
+    def value(self):
+        if not self.loaded and self.save_data is not None:
+            self.load()
+        if self.loaded:
+            return self.value_storage
+        else:
+            return self.default
+
+    def validate(self, value, account):
```

Comment:
Options aren't used only for Accounts. that's the default implementation, but you could just as easily use this for storing Options on a Global Script by adding the OptionHandler to it and providing an options_dict and a save category.

In this case, Account is passed because there are a rare few cases where the Account needs to be known. For instance, if a user is providing a DATETIME in their local time that the system needs to convert to UTC for storage. (This was a highly requested feature at MCM for the scene scheduler.)

Reply:
Sure, if it's stored on something other than the account, then that will also be accessible through the `.handler` property.

This seems to be a case of confused purposes. An Option describes the state of the object it sits on. It stores and handles that state. If you want to display that in an "looker aware" fashion, it seems this is something you should either handle after the option has been returned or (since I presume you want to make use of all that validation machinery) by passing it to a custom `display` method for that option.

Reply:
It's the other way around. Let's say I was to put this OptionHandler on a GlobalScript. now an admin wants to set a Datetime on one of its options. the admin is in Eastern Standard Time and wants to enter it in their local time. The system needs to convert this to UTC. How does the Option know which Account to pull from in determining the appropriate timezone to convert from?

Reply:
It's not up to the option to do this. The option stores the value you give it, in the type specified for that option. If the option is determined to be, say a UTC field, then the admin should supply their datetime with their local time zone embedded in that datetime object - the option will then convert that to whatever format it should hold.

Reply:
I will probably need your help coming up with a way for DUMB USERS to easily input dates without tripping over themselves, then, but okay. I'll remove the account parameters.